### PR TITLE
Add license data to SCA

### PR DESCRIPTION
**Pull request by maxamel**
## Title

This PR adds license info to the SCA report.

### Description
This PR calls an API to retrieve all license statuses for the relevant packages and displays them in the SCA report.

### License Statuses
Currently the license statuses are as follows: COMPLIANT, FAIL or empty.
COMPLIANT means all package licenses are compliant.
FAIL means some licenses are not compliant.
Empty means there is no data to determine compliance.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/sca_package/conftest.py [hunk 1]
```diff
@@ -203,7 +203,29 @@ def scan_result2() -> Dict[str, Any]:
             "medium": 2,
             "low": 0,
             "total": 6
-        }
+        },
+        'license_data': [{
+            'packageName': 'github.com/armon/go-metrics',
+            'packageVers ion': 'v0.0.0-20180917152333-f0300d1749da',
```
Comment (arielkru) [inline tests/sca_package/conftest.py:209]:
```suggestion
            'packageVersion': 'v0.0.0-20180917152333-f0300d1749da',
```

**Review by arielkru — APPROVED** [review]

**Review by NoaAzoulay — APPROVED** [review]

File: checkov/sca_package/runner.py [hunk 2]
```diff
@@ -162,6 +169,17 @@ def parse_vulns_to_records(
                     )
                 )
 
+    def get_license_status(self, license_dict, vulnerability):
```
Comment (gruebel) [inline checkov/sca_package/runner.py:172]: please add type hints here 🙂 probably something like that
```suggestion
    def get_license_status(self, license_dict: dict[tuple[str, str], dict[str, Any]], vulnerability: dict[str, Any]) -> str:
```

File: checkov/sca_package/scanner.py [hunk 3]
```diff
@@ -127,18 +127,30 @@ def run_scan_busy_wait(self, input_path: Path, scan_id: str) -> dict[str, Any]:
...
+        # handle license violations
+        request_body = {
+            "s3ObjectKey": scan_id
+        }
+        license_data = request_wrapper(
+            "POST", f"{self._base_url}/api/v1/vulnerabilities/packages/license-violations",
+            headers=bc_integration.get_default_headers("POST"),
+            json=request_body
+        )
+        license_data_json = license_data.json()
```
Comment (gruebel) [inline checkov/sca_package/scanner.py:146]: can the response be an empty dict/json? if yes, then the next line will blow up.

File: tests/sca_package/test_output.py [hunk 4]
```diff
@@ -380,20 +382,21 @@ def test_create_cli_output():
     # when
     cli_output = create_cli_output(True, records)
 
+    print(cli_output)
```
Comment (gruebel) [inline tests/sca_package/test_output.py:385]: looks like some debugging was happening here 😄 please remove

**Review by gruebel — COMMENTED** [review]
nice, added some comments

File: checkov/sca_package/scanner.py [hunk 5]
```diff
@@ -127,18 +127,30 @@ def run_scan_busy_wait(self, input_path: Path, scan_id: str) -> dict[str, Any]:
                 logging.error(response_json["outputData"])
                 return {}
 
+            time.sleep(SLEEP_DURATION)
```
Comment (schosterbarak) [inline checkov/sca_package/scanner.py:130]: why are we adding a sleep?

File: tests/sca_package/conftest.py [hunk 6]
```diff
@@ -1626,4 +1648,35 @@ def scan_result_success_response() -> Dict[str, Any]:
...
+
+@pytest.fixture()
+def license_violation_result_success_response() -> Dict[str, Any]:
+    return {
+        "packages": [
+            {
+                "packageName": "github.com/armon/go-metrics",
+                "packageVersion": "v0.0.0-20180917152333-f0300d1749da",
+                "packageLang": "go",
+                "license": "PRIVATE",
```
Comment (schosterbarak) [inline tests/sca_package/conftest.py:1662]: do we report MIT licenses or APACHE 2 licenses too?

Comment (schosterbarak) [reply tests/sca_package/conftest.py:1662]: the goal in SBOM is to have both compliant and non compliant licenses

File: checkov/sca_package/scanner.py (hunk 5 above)
Comment (maxamel) [reply checkov/sca_package/scanner.py:130]: @schosterbarak actually this sleep wasn't added , it just moved a few lines above. We sleep between calls in the original version as well.

File: checkov/sca_package/scanner.py (hunk 3 above)
Comment (maxamel) [reply checkov/sca_package/scanner.py:146]: @gruebel  the default response in case there's no data is {"packages": []} so should not crash in this case.

**Comment by maxamel** [comment]
Closing PR in order to move to a single API for fetching vulnerabilities and license violations. We'll also split the checkov output table to display vulnerabilities and license violations separately. @schosterbarak  @Saarett

**Pull request closed without merging**